### PR TITLE
Fixes for bin/run-browser-tests-local

### PR DESCRIPTION
**Pull request by nb1701**
This fixes several issues found with running `bin/run-browser-tests-local`.

1. Adds a `--local` flag to `bin/run-browser-test-env`, which sets `BASE_URL` to `http://localhost:9999`, which the local browser tests expect. Otherwise, upon logout, it will attempt to go to `civiform:9000` instead and fail. It also includes an additional compose file that exposes port 3390, as local tests will need to be able to talk to dev-oidc directly for redirects. Because this will conflict with `bin/run-dev`, this is not enabled by default.

2. Sets `LOCALSTACK_URL` to point to port 6645 in `bin/run-browser-tests-local`, which is what is specified in `browser-test-compose.yml`.

3. Adds the flag for starting the JVM debugger on port 9457 to allow connecting the debugger during browser tests.

4. Updates the `test_oidc_provider.js` file to add the paths for `localhost:9999` as allowed redirect URIs.

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by nb1701** [comment]
Still need to update docs

File: bin/run-browser-test-env
Comment (MichaelZetune) [inline bin/run-browser-test-env:58]: `local_mode`?

Comment (MichaelZetune) [inline bin/run-browser-test-env:58]: Is it possible for `--local` to be `--local_mode` to match? Or do they necessarily have to be different?

File: bin/run-browser-test-env
Comment (MichaelZetune) [inline bin/run-browser-test-env:104]: Add the `"$@"` here like in line 107

File: bin/run-browser-test-env
Comment (nb1701) [reply bin/run-browser-test-env:58]: No, they could be the same.  I'm not sure why I chose `local_mode` for the var name elsewhere :)

File: bin/run-browser-test-env
Comment (nb1701) [reply bin/run-browser-test-env:104]: Oops, good catch!

File: bin/run-browser-test-env
Comment (MichaelZetune) [inline bin/run-browser-test-env:59]: Suggestion: Since we have `bin/run-browser-tests` and bin/run-browser-tests-local`, can we have a `bin/run-browser-test-env-local` as well? Then we can remove the `--local` arg, set `BASE_URL` in `bin/run-browser-test-env-local`, and call `bin/run-browser-test-env` from that new file. What do you think?

File: bin/run-browser-test-env
Comment (MichaelZetune) [reply bin/run-browser-test-env:58]: Would you make that change? Unless you agree with my suggestion below.

File: bin/run-browser-test-env
Comment (nb1701) [reply bin/run-browser-test-env:59]: We could, but I think we'd end up duplicating a lot of the code in here?

Comment (nb1701) [reply bin/run-browser-test-env:59]: Sorry, didn't fully grok what you said at first.  Yeah, that's a good point, that might be cleaner.

File: browser-test/browser-test-compose.yml
Comment (MichaelZetune) [inline browser-test/browser-test-compose.yml:28]: For my own understanding, why does `BASE_URL` get declared twice here?

Comment (nb1701) [reply browser-test/browser-test-compose.yml:28]: This lets us override the `BASE_URL` env var in the container.  If it's not set, then we default it to `http://civiform:9000`.  This lets us set BASE_URL to something else (`http://localhost:9999`) for the local browser tests.

Comment (nb1701) [reply browser-test/browser-test-compose.yml:28]: I think to split it out into a separate file, we will either need to pass a flag into `run-browser-test-env` anyway to tell it to run in local mode or add something to `truth.sh` to detect when a variable is set and not exit, as we'll be wanting to set a variable in `run-browser-test-env-local`.  What do you think?

File: bin/run-browser-test-env
Comment (nb1701) [reply bin/run-browser-test-env:59]: I think to split it out into a separate file, we will either need to pass a flag into run-browser-test-env anyway to tell it to run in local mode or add something to truth.sh to detect when a variable is set and not exit, as we'll be wanting to set a variable in run-browser-test-env-local. What do you think?

Comment (MichaelZetune) [reply bin/run-browser-test-env:59]: I see - if we would need the `local_mode` flag regardless, then I'll leave it up to you. I guess it's not so different to run `bin/run-browser-test-env --local` vs `bin/run-browser-test-env-local`

**Comment by MichaelZetune** [comment]
Approved pending the unresolved comments and a quick test to make sure it's working after those changes. Thanks Nick!

**Review by MichaelZetune — APPROVED** [review]
Approved pending the unresolved comments and a quick test to make sure it's working after those changes. Thanks Nick!

File: bin/run-browser-test-env
Comment (nb1701) [reply bin/run-browser-test-env:59]: Alright.  I think for now I'll just leave it.  Suppose we could always have `bin/run-browser-test-env-local` that calls `bin/run-browser-test-env --local` :)

**Comment by nb1701** [comment]
Tested manually (with a locally rebuilt dev-oidc container) with the latest changes, everything still works 🎉

**Comment by nb1701** [comment]
Docs update: https://github.com/civiform/docs/pull/262

**Pull request merged**
